app/server/models/exchange.py

```diff
@@ -233,7 +228,9 @@ def exchange_from_amount(
             to_token,
             sender_transfer_account=find_transfer_accounts_with_matching_token(self.exchange_contract, to_token),
             recipient_user=user,
-            transfer_type=TransferTypeEnum.EXCHANGE, transfer_mode=transfer_mode
+            transfer_type=TransferTypeEnum.EXCHANGE,
+            transfer_mode=transfer_mode,
+            require_sufficient_balance=False
```

Comment:
these are just sprinkled in whereever we explicitly weren't checking for sufficient balances previously